servers/rendering/renderer_rd/storage_rd/render_scene_buffers_rd.h

```diff
@@ -0,0 +1,273 @@
+	_FORCE_INLINE_ Size2i get_target_size() const { return target_size; }
+	_FORCE_INLINE_ float get_fsr_sharpness() const { return fsr_sharpness; }
+	_FORCE_INLINE_ RS::ViewportMSAA get_msaa() const { return msaa; }
+	_FORCE_INLINE_ RS::ViewportScreenSpaceAA get_screen_space_aa() const { return screen_space_aa; }
+	_FORCE_INLINE_ bool get_use_taa() const { return use_taa; }
+	_FORCE_INLINE_ bool get_use_debanding() const { return use_debanding; }
+
+	uint64_t get_auto_exposure_version() const { return auto_exposure_version; }
+	void set_auto_exposure_version(const uint64_t p_auto_exposure_version) { auto_exposure_version = p_auto_exposure_version; }
+
+	// For our internal textures we provide some easy access methods.
+
+	_FORCE_INLINE_ RID get_internal_texture() const {
+		return get_texture(SNAME("RenderSceneBuffersRD"), SNAME("internal_texture"));
+	}
+	_FORCE_INLINE_ RID get_internal_texture(const uint32_t p_layer) {
+		return get_texture_slice(SNAME("RenderSceneBuffersRD"), SNAME("internal_texture"), p_layer, 0);
+	}
+
+	_FORCE_INLINE_ RID get_depth_texture() const {
+		return get_texture(SNAME("RenderSceneBuffersRD"), SNAME("depth_texture"));
+	}
+	_FORCE_INLINE_ RID get_depth_texture(const uint32_t p_layer) {
+		return get_texture_slice(SNAME("RenderSceneBuffersRD"), SNAME("depth_texture"), p_layer, 0);
+	}
+
+	// back buffer (color + depth)
+	RID get_back_buffer_texture() const { return blur[0].texture; } // We (re)use our blur texture here.
+
+	RID create_back_depth_texture();
+	_FORCE_INLINE_ RID get_back_depth_texture() { return has_texture(SNAME("RenderSceneBuffersRD"), SNAME("back_depth_texture")) ? get_texture(SNAME("RenderSceneBuffersRD"), SNAME("back_depth_texture")) : RID(); }
+	_FORCE_INLINE_ RID get_back_depth_texture(const uint32_t p_layer) { return has_texture(SNAME("RenderSceneBuffersRD"), SNAME("back_depth_texture")) ? get_texture_slice(SNAME("RenderSceneBuffersRD"), SNAME("back_depth_texture"), p_layer, 0) : RID(); }
+
+	// Velocity, currently only used by TAA (Clustered) but we'll be using this in other places soon too.
+
+	void ensure_velocity();
+	bool has_velocity_buffer(bool p_has_msaa) { return has_texture(SNAME("RenderSceneBuffersRD"), p_has_msaa ? SNAME("msaa_velocity") : SNAME("velocity")); }
+	RID get_velocity_buffer(bool p_get_msaa);
+	RID get_velocity_buffer(bool p_get_msaa, uint32_t p_layer) { return get_texture_slice(SNAME("RenderSceneBuffersRD"), p_get_msaa ? SNAME("msaa_velocity") : SNAME("velocity"), p_layer, 0); }
+
+	// VRS
+	_FORCE_INLINE_ bool has_vrs_texture() const { return has_texture(SNAME("VRS"), SNAME("Texture")); }
+	_FORCE_INLINE_ RID get_vrs_texture() const { return get_texture(SNAME("VRS"), SNAME("Texture")); }
+	_FORCE_INLINE_ RID get_vrs_texture(const uint32_t p_layer) { return get_texture_slice(SNAME("VRS"), SNAME("Texture"), p_layer, 0); }
+
+	////////////////////////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
I guess this will all move to generic API right?

Reply:
The problem with VRS is that it needs to be used when setting up frame buffers and is thus handled by the main class. I want to give this some more thought how we can better have the implementation just in the VRS effects class.

Reply:
Ok I changed how this works so we not longer have these getters. Where VRS is used it just accesses the buffers directly.